pkg/updatestatus/nodeinformer.go

```diff
@@ -194,6 +202,37 @@ func (c *nodeInformerController) sync(ctx context.Context, syncCtx factory.SyncC
 	return nil
 }
 
+func (c *nodeInformerController) initializeCachesNoErrors() {
+	if err := c.initializeCaches(); err != nil {
+		klog.Errorf("Failed to initialize caches: %v", err)
+	}
+}
+
+func (c *nodeInformerController) initializeCaches() error {
+	var errs []error
+
+	if pools, err := c.machineConfigPools.List(labels.Everything()); err != nil {
+		errs = append(errs, err)
+	} else {
+		for _, pool := range pools {
+			c.machineConfigPoolSelectorCache.ingest(pool)
+		}
+	}
+	klog.V(2).Infof("Stored %d machineConfigPools in the cache", c.machineConfigPoolSelectorCache.len())
```

Comment:
consider not having a `len()` (that needs to lock) and just count the ingestions?

Reply:
Done.

https://github.com/openshift/cluster-version-operator/pull/1164/commits/a81ec69e31f504ce10a44650cfa5c4d4260fd2cd#diff-c5a41cb83200e13544e0509a5e77c180e16a1d70d7411a79cac2bc107e89950fR155